### PR TITLE
Add syslog exporter docs

### PR DESCRIPTION
**Pull request by attilakreiner**
Add documentation page for syslog exporter.

### CODE REVIEW

File: src/reference/config/telemetry/exporters/exporter-syslog.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+#### options.protocol
+
+> `string`
+
+Protocol is the protocol to use to communicate with the syslog server. Valid values are: `tcp`, `udp`, `tls`.
+
+#### options.trust
+
+> `array` of `strings`
+
```
Comment (jfallows) [inline src/reference/config/telemetry/exporters/exporter-syslog.md:94]: These are not really trust files. 

These are keys into the vault referenced on the binding, which could be a filesystem vault for a local pkcs12 keystore or an aws vault for remote pem format certificates stored in AWS secrets manager.

Perhaps we need to reference vault in the description and example here?

**Review by jfallows — CHANGES_REQUESTED** [review]

File: src/reference/config/telemetry/exporters/exporter-syslog.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  protocol: tls
+  trust:
+    - syslog
+```
+
+#### options.host
+
+> `string`
+
+Host is the hostname of the syslog server.
```
Comment (vordimous) [inline src/reference/config/telemetry/exporters/exporter-syslog.md:77]:
```suggestion
The hostname of the syslog server.
```

File: src/reference/config/telemetry/exporters/exporter-syslog.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+#### options.port
+
+> `integer`
+
+Port is the port of the syslog server.
```
Comment (vordimous) [inline src/reference/config/telemetry/exporters/exporter-syslog.md:83]:
```suggestion
The port of the syslog server.
```

File: src/reference/config/telemetry/exporters/exporter-syslog.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+#### options.protocol
+
+> `string`
+
+Protocol is the protocol to use to communicate with the syslog server. Valid values are: `tcp`, `udp`, `tls`.
```
Comment (vordimous) [inline src/reference/config/telemetry/exporters/exporter-syslog.md:89]:
```suggestion
The protocol to use to communicate with the syslog server. Valid values are: `tcp`, `udp`, `tls`.
```

**Review by vordimous — CHANGES_REQUESTED** [review]

File: src/reference/config/telemetry/exporters/exporter-syslog.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+> `enum` [ "tcp", "udp", "tls" ] | Default: `"tcp"`
+
+The protocol to use to communicate with the syslog server. Valid values are: `tcp`, `udp`, `tls`.
+
+#### options.trust
+
+> `array` of `string`
+
+Keys in the vault referenced on the binding (e.g. a filesystem vault for a local pkcs12 keystore
+or an AWS vault for remote pem format certificates stored in AWS secrets manager). Only valid if the protocol is `tls`.
```
Comment (jfallows) [inline src/reference/config/telemetry/exporters/exporter-syslog.md:118]: Let's use the tls binding as a guide on how to document this.
https://github.com/aklivity/zilla-docs/blob/develop/src/reference/config/bindings/binding-tls.md

The first example probably needs to include a vault name and the tls configuration.

**Review by jfallows — CHANGES_REQUESTED** [review]

**Review by vordimous — APPROVED** [review]

File: src/reference/config/telemetry/exporters/exporter-syslog.md (hunk 5 above)
Comment (attilakreiner) [reply src/reference/config/telemetry/exporters/exporter-syslog.md:125]: I added the missing doc entry for the vault param.

I have to examples on the top of the page, one without vault, and one with vault, and tls like this
```yaml {3}
exporters:
  syslog:
    type: syslog
    options:
      host: syslog-server
      port: 514
      protocol: tcp
```

with vault:
```
vaults:
  vault:
    type: filesystem
    options:
      trust:
        store: truststore.p12
        type: pkcs12
        password: generated
exporters:
  syslog:
    type: syslog
    vault: vault
    options:
      host: syslog-server
      port: 6514
      protocol: tls
      trust:
        - syslog
```

Pls LMK if this is OK or if we need something more.

File: src/reference/config/telemetry/exporters/exporter-syslog.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+::::
+
+### vault
+
+> `string`
+
+Vault name. Only applicable if the protocol is `tls`.
+
+### options
```
Comment (vordimous) [inline src/reference/config/telemetry/exporters/exporter-syslog.md:73]:
```suggestion
### options*
```

File: src/reference/config/telemetry/exporters/exporter-syslog.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+```yaml {4}
+options:
+  host: syslog-server
+  port: 6514
+  protocol: tls
+  trust:
+    - syslog
+```
+
+#### options.host
```
Comment (vordimous) [inline src/reference/config/telemetry/exporters/exporter-syslog.md:102]:
```suggestion
#### options.host*
```

File: src/reference/config/telemetry/exporters/exporter-syslog.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+> `string`
+
+The hostname of the syslog server.
+
+#### options.port
```
Comment (vordimous) [inline src/reference/config/telemetry/exporters/exporter-syslog.md:108]:
```suggestion
#### options.port*
```

File: src/reference/config/telemetry/exporters/exporter-syslog.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+
+> `integer`
+
+The port of the syslog server.
+
+#### options.protocol
```
Comment (vordimous) [inline src/reference/config/telemetry/exporters/exporter-syslog.md:114]:
```suggestion
#### options.protocol*
```

**Review by vordimous — CHANGES_REQUESTED** [review]

File: src/reference/config/telemetry/exporters/exporter-syslog.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+exporters:
+  syslog:
+    type: syslog
+    vault: vault
+    options:
+      host: syslog-server
+      port: 6514
+      protocol: tls
+      trust:
+        - syslog
```
Comment (vordimous) [inline src/reference/config/telemetry/exporters/exporter-syslog.md:46]:
```suggestion
vaults:
  my_syslog_trust:
    type: filesystem
    options:
      trust:
        store: truststore.p12
        type: pkcs12
        password: generated
exporters:
  syslog:
    type: syslog
    vault: my_syslog_trust
    options:
      host: syslog-server
      port: 6514
      protocol: tls
      trust:
        - syslog
```

**Review by vordimous — CHANGES_REQUESTED** [review]

**Review by vordimous — APPROVED** [review]

**Review by jfallows — APPROVED** [review]

**Pull request merged**